File: abc337a/abc337a.hs
import Control.Monad(replicateM)
import Data.List(transpose)

main :: IO ()
main = do
  n <- readLn
  solve <$> replicateM n f >>= putStrLn
  where
    f = map read <$> words <$> getLine

solve :: [[Int]] -> String
solve = f . map sum . transpose
  where
    f [a, b] = case compare a b of
      GT -> "Takahashi"
      LT -> "Aoki"
      EQ -> "Draw"

File: abc337a/abc337a.ml
open Printf
open Scanf

let solve l =
  let rec loop l tk ao = match l with
      (x, y) :: r -> loop r (tk + x) (ao + y)
    | _ -> if tk > ao then "Takahashi" else if tk < ao then "Aoki" else "Draw" in
  loop l 0 0

let () =
  scanf "%d " (fun n ->
    List.init n (fun _ -> scanf "%d %d " @@ fun x y -> (x, y))) |> solve |> printf "%s\n"
